### PR TITLE
Add user positions endpoint with tests

### LINKED ISSUE #351: [Backend] Create endpoint to return position history
🚨 Attention: Please read the guidelines before applying; otherwise, you will be ignored. 🚫📋
0. Please, join the telegram group to keep in touch https://t.me/+m0IGRiBvLtsyMzIy

1. Read guideline [here](https://gist.github.com/djeck1432/f722f5fe518c80b248ff609627da25e5)
2. Write in your application ETA.
3. Create endpoint in [position.py](https://github.com/djeck1432/spotnet/blob/main/web_app/api/position.py) to return all user positions by wallet_id
4. Add Response model
5. Add crud method to [PositionDBConnector](https://github.com/djeck1432/spotnet/blob/main/web_app/db/crud/position.py#L21)
6. Add test case for this endpoint

### PR DESCRIPTION
**Pull request by akintewe**
## Changes
- Add GET `/api/user-positions/{wallet_id}` endpoint to fetch all positions for a user
- Create response models `UserPositionResponse` and `UserPositionsListResponse`
- Implement `get_positions_by_wallet_id` in PositionDBConnector
- Add test cases for success and error scenarios

## Testing
- Added unit tests for:
  - Successful position retrieval
  - Empty wallet ID handling
  - No positions case
  - Invalid wallet ID case

Related Issues
Closes #351

### CODE REVIEW

File: web_app/api/serializers/position.py [hunk 1]
```diff
@@ -55,3 +57,24 @@ class Config:
...
+    Represents a single position in the user's position list.
+    """
+    id: str
+    token_symbol: str
+    amount: str
+    multiplier: float
+    status: str
+    created_at: datetime
+    start_price: float
+    is_liquidated: bool
```
Comment (djeck1432) [inline web_app/api/serializers/position.py:73]: please, add `datetime_liquidation`

File: web_app/api/serializers/position.py [hunk 2]
```diff
@@ -55,3 +57,24 @@ class Config:
...
+
+
+class UserPositionsListResponse(BaseModel):
+    """
+    Response model for list of user positions
+    """
+    positions: List[UserPositionResponse]
```
[the remark this answers is not in the document]
Comment (akintewe) [reply web_app/api/serializers/position.py:80]: ok noted will be modified now

**Comment by akintewe** [comment]
@djeck1432  Please verify now

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**